src/tabs/newTab.tsx

```diff
@@ -1,16 +1,67 @@
+import { useEffect, useState } from "react"
+
+import "./NewTab.css"
+
+import { BlocksRenderer } from "@strapi/blocks-react-renderer"
+import Browser from "webextension-polyfill"
+
+import { SlimPopup } from "~types/eventTypes"
+
 const NewTab = () => {
   const params = new URLSearchParams(window.location.search)
-  const width = params.get("width")
-  const height = params.get("height")
-  const url = "http://localhost:1337" + params.get("url")
-
+  const index = params.get("index")
+  const [popup, setPopup] = useState<SlimPopup>()
+  useEffect(() => {
+    const getFromStorage = async () => {
+      const { popups } = await Browser.storage.session.get(["popups"])
+      setPopup(popups[index])
+    }
+    getFromStorage()
+  }, [])
   return (
-    <div>
-      <img
+    <div style={{ width: "100%", height: "100%" }}>
+      {popup && (
+        <>
+          {popup.type === "text" && (
+            <>
+              <h1>{popup.popupInfo.work_title}</h1>
+              <BlocksRenderer content={popup.text_content} />
+              <h1>{popup.popupInfo.work_title}</h1>
+              <BlocksRenderer content={popup.text_content} />
+            </>
+          )}
+          {popup.type === "image" && (
+            <img
+              src={
+                process.env.NODE_ENV === "development"
+                  ? "http://localhost:1337" + popup.url
+                  : popup.url
```

Comment:
this is because of that issue where strapi manipulates the url returned differently between `dev` and `production`?

Reply:
Yes exactly. When the url will come from the production deployed cms it will be a complete url to the stored image url in the bucket. Mostly put this here as reminder to myself